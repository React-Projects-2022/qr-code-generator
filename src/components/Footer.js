export const Footer = () => (
  <div class="footer">
    <p>QR Code Generator - Anartz Mugika Ledo - { new Date().getFullYear()}</p>
  </div>
);
