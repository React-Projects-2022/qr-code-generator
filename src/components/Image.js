export const Image = ({ qrCode, text }) => (
  <img src={qrCode} alt={"QR Code Image with ".concat(text)} />
);
