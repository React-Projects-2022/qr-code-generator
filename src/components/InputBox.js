export const InputBox = ({
  size,
  temp,
  changeSize,
  changeBgColor,
  asignInputTempText,
  asignFindText,
}) => {
  const handleClick = () => asignFindText(temp);
  return (
    <div className="input-box">
      <div className="gen">
        <input
          type="text"
          autoFocus={true}
          onChange={(e) => {
            asignInputTempText(e.target.value);
          }}
          placeholder="Enter text to encode"
        />
        <button className="button" onClick={handleClick}>
          Generate
        </button>
      </div>
      <div className="extra">
        <h5>Background Color:</h5>
        <input
          type="color"
          onChange={(e) => {
            changeBgColor(e.target.value.substring(1));
          }}
        />
        <h5>Dimension:</h5>
        <input
          type="range"
          min="200"
          max="600"
          value={size}
          onChange={(e) => {
            changeSize(e.target.value);
          }}
        />
      </div>
    </div>
  );
};
