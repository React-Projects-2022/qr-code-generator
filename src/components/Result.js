import { Image } from "./Image";
import { downloadImage } from "./../helpers/download-image";
export const Result = ({ qrCode, text }) => (
  <div className="output-box">
    <Image qrCode={qrCode} text={text} />
    <button onClick={() => downloadImage(qrCode, text)} type="button">
      Download
    </button>
  </div>
);
