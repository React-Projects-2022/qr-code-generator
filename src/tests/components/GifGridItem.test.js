import { GifGridItem } from "../../components/GifGridItem";
import { shallow } from 'enzyme';
describe('Pruebas en GifGridItem', () => {
    let img;
    let wrapper;
    beforeEach(() => {
        img = {
            title: 'Reaction',
            url: 'https://media0.giphy.com/media/tfUW8mhiFk8NlJhgEh/giphy.gif?cid=88be9d0a1nf5v24mivlell2rqw0r5g8tt54n7i76j175j9xk&rid=giphy.gif&ct=g'
        }
        wrapper = shallow(<GifGridItem img={img}/>);
    })
    test('Debe de mostrar correctamente el componente', () => {
        
        expect( wrapper ).toMatchSnapshot();
    });

    test('Comprobar si el título es el correcto', () => {
        const title = wrapper.find("p").text().trim();
        expect(title).toBe(img.title);
    });

    test('Comprobar si propiedades de imagen son correctas', () => {
        
        const imgData = wrapper.find("img").props();

        expect(imgData.src).toBe(img.url);
        expect(imgData.alt).toBe(img.title);
    });

    test('Comprobar si está añadiendo correctamente clase "animate__fadeIn"', () => {
        
        const divClassName = wrapper.find("div").props().className;
        expect(divClassName.includes('animate__fadeIn')).toBe(true);
    
    });
});