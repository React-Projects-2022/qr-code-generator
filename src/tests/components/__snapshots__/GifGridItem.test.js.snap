// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de mostrar correctamente el componente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Reaction"
    src="https://media0.giphy.com/media/tfUW8mhiFk8NlJhgEh/giphy.gif?cid=88be9d0a1nf5v24mivlell2rqw0r5g8tt54n7i76j175j9xk&rid=giphy.gif&ct=g"
  />
  <p>
    Reaction
  </p>
</div>
`;
